clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -435,6 +437,15 @@ public static Integer getPort(String address) {
         return matcher.matches() ? Integer.parseInt(matcher.group(2)) : null;
     }
 
+    /**
+     * Basic validation of the supplied address. checks for valid characters
+     * @param address
+     * @return true if address contains valid characters
+     */
+    public static boolean validHostPattern(String address) {
```

Comment:
Inside `Utils.parseAndValidateAddresses`, we log a warning if the host is not resolved. Do you think it's worthwhile doing the same thing here?

Reply:
not sure if it required here. For any unresolvable/invalid format will throw error during server startup [here](https://github.com/apache/kafka/pull/5334/files#diff-24ec8055938739c40d2c11cafff35455R889) and during dynamic config update [here](https://github.com/apache/kafka/pull/5334/files#diff-24ec8055938739c40d2c11cafff35455R907)